### DISCUSSION TITLE
Welcome to azure-search-openai-demo Discussions!

### DISCUSSION THREAD

**Discussion by pamelafox** [Azure-Samples/azure-search-openai-demo#2239] [Announcements]
## 👋 Welcome!

We're enabling discussions in this repo since we often get questions that aren't necessarily bugs or feature requests, but are good topics for discussion, and we want to bring together all of you developers that are working on RAG solutions.

Please make sure that if you are experiencing any sort of issue, you first search the issue tracker, as quite a few issues have been discussed there.

Feel free to introduce yourself to the community in this thread!

**Comment by pamelafox** [comment]
To introduce myself: I'm Pamela, one of the primary maintainers. I'm on the Python cloud advocacy team at Microsoft and maintain quite a few Python templates to make it easy to deploy common architectures and frameworks. You can see a list here: https://github.com/pamelafox
I enjoy working on this RAG repo, since RAG is such a powerful use of LLMs, and I love learning from all of you about what works and doesn't work when you deploy RAG on your domain.

I'll be mostly away for the next few weeks, due to the holidays, but I'll be back in full force in January and putting on a 10-part live stream series to dive deep into this repo: https://developer.microsoft.com/en-us/reactor/series/S-1450/

**Reply by jrichMQ** [reply]
Hi Pamela - I'm working to deploy this Rag repo, with some customizations at my organization. Thank you for putting this together and for all the content! Excited to watch and learn from the deep dive series. 

Are there ways I can contribute to this repo? If so, how can I go about doing that to increase my learning and experience working with python and open source coding?

Thanks,
Jake

**Comment by KumarAman02** [comment]
Why are we using Azure services, can't we use AWS services? Is there any specific reason for this?

**Reply by pamelafox** [reply]
This sample was created by Microsoft employees to demonstrate how well Azure AI Search works for RAG applications. AWS may have similar code solutions, I do not know since I work on Azure.

**Comment by sanjoyganguly** [comment]
@pamelafox Is possible to share the low level architecture diagram ? Also, what are the minimum Azure services required to run the web application. ( Eg. I do not want to use cosmos DB , vision right now but may be later on )

Thanks in advance for your help.

**Reply by pamelafox** [reply]
This is what is provisioned by default:

- Azure Container Apps: Default host for app deployment as of 10/28/2024. See more details in [the ACA deployment guide](docs/azure_container_apps.md). Consumption plan with 1 CPU core, 2.0 GB RAM. Pricing with Pay-as-You-Go. [Pricing](https://azure.microsoft.com/pricing/details/container-apps/)
- Azure Container Registry: Basic tier. [Pricing](https://azure.microsoft.com/pricing/details/container-registry/)
- Azure OpenAI: Standard tier, GPT and Ada models. Pricing per 1K tokens used, and at least 1K tokens are used per question. [Pricing](https://azure.microsoft.com/pricing/details/cognitive-services/openai-service/)
- Azure AI Document Intelligence: SO (Standard) tier using pre-built layout. Pricing per document page, sample documents have 261 pages total. [Pricing](https://azure.microsoft.com/pricing/details/form-recognizer/)
- Azure AI Search: Basic tier, 1 replica, free level of semantic search. Pricing per hour. [Pricing](https://azure.microsoft.com/pricing/details/search/)
- Azure Blob Storage: Standard tier with ZRS (Zone-redundant storage). Pricing per storage and read operations. [Pricing](https://azure.microsoft.com/pricing/details/storage/blobs/)
- Azure Monitor: Pay-as-you-go tier. Costs based on data ingested. [Pricing](https://azure.microsoft.com/pricing/details/monitor/)

Here's a diagram:
![Screenshot 2025-02-11 at 8 55 48 AM](https://github.com/user-attachments/assets/68f252dc-0693-44bd-bc09-11858f2264ef)


There's a lot more that can be added on, and App Service can be swapped with Container Apps.

**Comment by sheetal37** [comment]
I want to call your backend API, so how should I do it?

**Reply by pamelafox** [reply]
I've added documentation detailing the HTTP protocol:
https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/docs/http_protocol.md
Let me know if that's unclear.

**Reply by sheetal37** [reply]
Thanks for the update! The HTTP protocol doc is clear — appreciate it!
